### ISSUE TITLE
Consider how to support spring.integration.keep-alive

### ISSUE THREAD

**Issue by philwebb** [spring-projects/spring-boot#42469]
Spring Integration is introducing a new `spring.integration.keep-alive` property which will create a keep-alive thread [based on this logic](https://docs.spring.io/spring-integration/reference/6.4-SNAPSHOT/keep-alive.html). We could surface this property in Spring Boot, however, we already have `spring.main.keep-alive`.

We need to consider what we should do by default, specifically:

* If `spring.main.keep-alive` is `true` then there is no need to Spring Integration to create a thread
* If `spring.main.keep-alive` is missing then standard Spring Integration logic should apply
* If `spring.main.keep-alive`is `false` we may need to also disable Spring Integration keep alive
* If we don't do the above, we need to offer a way to turn off Spring Integration keep alive threads

**Comment by philwebb** [comment]
See https://github.com/spring-projects/spring-integration/issues/9455 for the original request. We need some more time to consider our options portfolio wide so we'll need to look at this after 3.4

**Comment by mhalbritter** [comment]
This feature has been introduced in https://github.com/spring-projects/spring-integration/pull/9493, but then has been reverted in https://github.com/spring-projects/spring-integration/commit/2cf2f109f97e23f7af9dc07f14e6b701d6da1018.

I couldn't find the keep alive feature on their `main` branch, so I think it's gone for good.

@artembilan is that assumption correct?

**Comment by artembilan** [comment]
That's correct, @mhalbritter . The feature was introduced and I tried to incorporate it into Spring Boot.
But in the end we decided with @philwebb to revert it in favor of existing in Spring Boot and idea to come up with some more general solution.

Not sure where we are, but this issue is still opened: https://github.com/spring-projects/spring-integration/issues/9455.
